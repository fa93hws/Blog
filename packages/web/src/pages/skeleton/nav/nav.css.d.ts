// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'authorContainer': string;
  'authorMarginLeft': string;
  'avatarImg': string;
  'colorBlueUltraLight': string;
  'colorPurpleLight': string;
  'content': string;
  'email': string;
  'gridBaseline': string;
  'monoGreyA200': string;
  'monoGreyLight': string;
  'monoWhite': string;
  'monoWhiteA50': string;
  'name': string;
  'nav': string;
  'sourceCodeBanner': string;
  'sourceCodeBannerHeight': string;
  'sourceCodeBannerWidth': string;
}
export const cssExports: CssExports;
export default cssExports;
