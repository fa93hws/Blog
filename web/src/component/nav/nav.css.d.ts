// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'avatar': string;
  'content': string;
  'nav': string;
}
export const cssExports: CssExports;
export default cssExports;
