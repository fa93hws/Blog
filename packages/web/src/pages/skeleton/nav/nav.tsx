import * as React from 'react';
import { Text } from 'components/typography/typography';
import styles from './nav.css';
import { NavItem, NavLinks } from './nav-links/nav-links';

export const Nav = ({ navItems }: { navItems: readonly NavItem[] }) => (
  <nav className={styles.nav}>
    <div className={styles.sourceCodeBanner}>
      <a
        href="https://github.com/fa93hws/blog"
        target="_blank"
        rel="noopener noreferrer"
      >
        <Text.Micro>源代码</Text.Micro>
      </a>
    </div>
    <div className={styles.authorContainer}>
      <a
        href="https://github.com/fa93hws/"
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          src="https://avatars0.githubusercontent.com/u/10626756"
          alt="avatar"
          className={styles.avatarImg}
        />
      </a>
      <div className={styles.name}>
        <Text.Micro>夏目天子</Text.Micro>
      </div>
      <div className={styles.email}>
        <Text.Micro>wjun0912@gmail.com</Text.Micro>
      </div>
    </div>
    <NavLinks items={navItems} />
  </nav>
);
