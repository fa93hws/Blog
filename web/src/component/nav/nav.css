.nav {
  display: flex;
  flex-direction: column;
}

.content {
  flex-grow: 7;
  background-color: rgb(44, 35, 35);
}
