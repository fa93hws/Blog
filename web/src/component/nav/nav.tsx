import * as React from 'react';
import styles from './nav.css';
import { Author, AuthorProps } from './author/author';

export const Nav = ({ author }: { author: AuthorProps }) => (
  <nav className={styles.nav}>
    <Author name={author.name} email={author.email} avatar={author.avatar} />
    <div className={styles.content}>This is content</div>
  </nav>
);
