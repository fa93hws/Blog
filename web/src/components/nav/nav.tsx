import * as React from 'react';
import styles from './nav.css';
import { Author, AuthorProps } from './author/author';
import { SourceCodeBanner } from './banner/banner';
import { NavItem, NavLinks } from './nav-links/nav-links';

export const Nav = ({
  author,
  sourceCodeUrl,
  navItems,
}: {
  author: AuthorProps;
  sourceCodeUrl: string;
  navItems: readonly NavItem[];
}) => (
  <nav className={styles.nav}>
    <SourceCodeBanner sourceCodeUrl={sourceCodeUrl} />
    <Author name={author.name} email={author.email} avatar={author.avatar} />
    <NavLinks items={navItems} />
  </nav>
);
