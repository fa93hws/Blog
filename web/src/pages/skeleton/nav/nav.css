@value gridBaseline from '~components/metrics.css';
@value colorBlueUltraLight from '~components/colors.css';
@value colorPurpleLight from '~components/colors.css';
@value monoWhite, monoWhiteA50 from '~components/colors.css';
@value monoGreyLight, monoGreyA200 from '~components/colors.css';

@value authorMarginLeft: calc(gridBaseline * 3);
@value sourceCodeBannerWidth: calc(gridBaseline * 16);
@value sourceCodeBannerHeight: calc(gridBaseline * 4);

.nav {
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.content {
  flex-grow: 7;
  background-color: rgb(44, 35, 35);
}

.authorContainer {
  background-color: colorBlueUltraLight;
  padding: calc(gridBaseline * 5) 0;
}

.avatarImg {
  width: calc(gridBaseline * 10);
  height: calc(gridBaseline * 10);
  border-radius: 50%;
  margin-left: authorMarginLeft;
}

.name {
  white-space: nowrap;
  color: monoWhite;
  margin-top: gridBaseline;
  margin-left: authorMarginLeft;
}

.email {
  white-space: nowrap;
  color: colorPurpleLight;
  margin-left: authorMarginLeft;
}

.sourceCodeBanner {
  width: sourceCodeBannerWidth;
  height: sourceCodeBannerHeight;
  left: calc(sourceCodeBannerWidth * 0.707 / 2 - sourceCodeBannerWidth / 2);
  top: calc(sourceCodeBannerWidth * 0.707 / 2 - sourceCodeBannerHeight / 2);
  transform: rotate(-45deg) translate(0, calc(0 - sourceCodeBannerHeight));

  position: absolute;
  overflow: hidden;
  z-index: 15;
  text-align: center;
  background-color: green;
  box-shadow: 0 0 calc(gridBaseline * 6) 0 monoGreyA200;
  color: monoWhiteA50;
  cursor: pointer;

  display: flex;
  justify-content: center;
  align-items: center;
}

.sourceCodeBanner:hover {
  text-decoration: underline;
  box-shadow: 0 0 calc(gridBaseline * 2) gridBaseline monoGreyLight;
}
