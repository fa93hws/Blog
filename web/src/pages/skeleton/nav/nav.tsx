import * as React from 'react';
import { Link } from 'react-router-dom';
import { Text } from 'components/typography/typography';
import styles from './nav.css';
import { NavItem, NavLinks } from './nav-links/nav-links';

export const Nav = ({ navItems }: { navItems: readonly NavItem[] }) => (
  <nav className={styles.nav}>
    <div className={styles.sourceCodeBanner}>
      <Link to="https://github.com/fa93hws/blog">
        <Text.Micro>源代码</Text.Micro>
      </Link>
    </div>
    <div className={styles.authorContainer}>
      <Link to="https://github.com/fa93hws/">
        <img
          src="https://avatars0.githubusercontent.com/u/10626756"
          alt="avatar"
          className={styles.avatarImg}
        />
      </Link>
      <div className={styles.name}>
        <Text.Micro>夏目天子</Text.Micro>
      </div>
      <div className={styles.email}>
        <Text.Micro>wjun0912@gmail.com</Text.Micro>
      </div>
    </div>
    <NavLinks items={navItems} />
  </nav>
);
