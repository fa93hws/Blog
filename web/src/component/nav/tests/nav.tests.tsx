import * as React from 'react';
import { Nav } from '../nav';

describe('Global Nav', () => {
  it('should match the snapshot', () => {
    const author = {
      name: 'author name',
      email: 'author email',
      avatar: {
        img: 'author avatar path',
        alt: 'author avatar alt',
        linkTo: 'author avatar link',
      },
    };
    expect(<Nav author={author} />).toMatchRenderedSnapshot();
  });
});
