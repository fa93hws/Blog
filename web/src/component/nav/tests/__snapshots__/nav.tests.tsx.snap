// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Nav should match the snapshot 1`] = `
<nav
  class="nav"
>
  <div
    class="container"
  >
    <a
      href="author avatar link"
    >
      <img
        alt="author avatar alt"
        class="avatarImg"
        src="author avatar path"
      />
    </a>
    <p
      class="name"
    >
      author name
    </p>
    <p
      class="email"
    >
      author email
    </p>
  </div>
  <div
    class="content"
  >
    This is content
  </div>
</nav>
`;
